src/linter.ts

```diff
@@ -19,13 +19,18 @@ const isAKnownException = (violation: IRuleResult, locations: IExceptionLocation
       continue;
     }
 
-    for (let i = 0; i < violation.path.length; i++) {
+    let arePathEquals: boolean = true;
```

Comment:
Dammit. Those ten lines of code were wrong is so many ways...

Thanks @m-mohr for all the help and feedback!